server/src/main/java/org/elasticsearch/action/admin/cluster/allocation/TransportGetAllocationStatsAction.java

```diff
@@ -185,4 +213,42 @@ public DiskThresholdSettings getDiskThresholdSettings() {
             return diskThresholdSettings;
         }
     }
+
+    private record CachedAllocationStats(Map<String, NodeAllocationStats> stats, long timestampMsecs) {}
+
+    private static class AllocationStatsCache {
+        private volatile long maxAgeMsecs;
```

Comment:
Also, naming nit, I'd suggest `maxAgeMillis` - we use `Millis` as a suffix in lots of other time-related places, and `Msecs` reads as megaseconds to me :)